runtime/vm/resolvesupport.cpp

```diff
@@ -2104,6 +2278,69 @@ j9object_t
 resolveInvokeDynamic(J9VMThread *vmThread, J9ConstantPool *ramCP, UDATA callSiteIndex, UDATA resolveFlags)
 {
 	Assert_VM_true(J9_RESOLVE_FLAG_RUNTIME_RESOLVE == resolveFlags);
+#if defined(J9VM_OPT_OPENJDK_METHODHANDLE)
+	J9InvokeCacheEntry *invokeCache = (J9InvokeCacheEntry *)ramCP->ramClass->callSites + callSiteIndex;
+	j9object_t result = invokeCache->target;
+
+	J9ROMClass *romClass = ramCP->ramClass->romClass;
+	J9SRP *callSiteData = (J9SRP *) J9ROMCLASS_CALLSITEDATA(romClass);
+	U_16 *bsmIndices = (U_16 *) (callSiteData + romClass->callSiteCount);
+	U_16 *bsmData = bsmIndices + romClass->callSiteCount;
+	J9ROMNameAndSignature* nameAndSig = SRP_PTR_GET(callSiteData + callSiteIndex, J9ROMNameAndSignature*);
+	U_16 bsmIndex = bsmIndices[callSiteIndex];
+	U_16 i;
+
+	/* Check if already resolved */
+	if (result != NULL) {
+		return result;
+	}
+
+	/* Walk bsmData - skip all bootstrap methods before bsmIndex */
+	for (i = 0; i < bsmIndex; i++) {
+		/* increment by size of bsm data plus header */
+		bsmData += bsmData[1] + 2;
+	}
+
+	sendResolveInvokeDynamic(vmThread, ramCP, callSiteIndex, nameAndSig, bsmData);
+	result = (j9object_t)vmThread->returnValue;
+
+	/* check if an exception is already pending */
+	if (vmThread->currentException != NULL) {
+		/* Already a pending exception */
+		result = NULL;
+	} else if (result == NULL) {
+		setCurrentExceptionUTF(vmThread, J9VMCONSTANTPOOL_JAVALANGNULLPOINTEREXCEPTION, NULL);
+	}
+
+	/* Only write the value in if its not null */
+	if (NULL != result) {
+		j9object_t memberName = (j9object_t)J9JAVAARRAYOFOBJECT_LOAD(vmThread, result, 0);
+		j9object_t appendix = (j9object_t)J9JAVAARRAYOFOBJECT_LOAD(vmThread, result, 1);
+		J9MemoryManagerFunctions *gcFuncs = vmThread->javaVM->memoryManagerFunctions;
+		memberName = gcFuncs->j9gc_objaccess_asConstantPoolObject(
+									vmThread,
+									memberName,
+									J9_GC_ALLOCATE_OBJECT_TENURED | J9_GC_ALLOCATE_OBJECT_NON_INSTRUMENTABLE | J9_GC_ALLOCATE_OBJECT_HASHED);
+		appendix = gcFuncs->j9gc_objaccess_asConstantPoolObject(
+									vmThread,
+									appendix,
+									J9_GC_ALLOCATE_OBJECT_TENURED | J9_GC_ALLOCATE_OBJECT_NON_INSTRUMENTABLE | J9_GC_ALLOCATE_OBJECT_HASHED);
```

Comment:
What kind of objects are `memberName` & `appendix`?  Do their identities matter?  If so we can't do this as these APIs create a copy in tenure space, changing the identity.

ie: For the MethodTypes, we need to ensure that they are `==` pointer-equals so cloning to tenure here would break that